ui/src/app/login/components/utils.ts

```diff
@@ -153,7 +160,18 @@ export const pkceCallback = async (queryParams: string, oidcConfig: AuthSettings
         throw new PKCELoginError('No token in response');
     }
 
-    document.cookie = `argocd.token=${result.id_token}; path=/`;
+    // This regex removes any leading or trailing '/' characters and the result is appended to a '/'.
+    // This is because when base href if not just '/' toAbsURL() will append a trailing '/'.
+    // Just removing a trailing '/' from the string would break when base href is not specified, defaulted to '/'.
+    // This pattern is used to handle both cases.
+    document.cookie = `argocd.token=${result.id_token}; path=/${requests.toAbsURL('').replace(/^\/|\/$/g, '')}`;
 
-    window.location.replace('/applications');
+    const returnURL = sessionStorage.getItem('return_url');
+
+    if (returnURL) {
+        sessionStorage.removeItem('return_url');
```

Comment:
Can you describe a little more the flow here? I see that the return_url is added to the sessionStorage and then removed from it. What would happen after setting it and before/after removing it?

Reply:
Once a PKCE auth flow has been initiated, I attempt to cache the path that the user is at to sessionStorage. This caches the path information locally on the users browser before the user is redirected to the OIDC provider to authenticate with PKCE.

Once the user is authenticated and redirected back to ArgoCD with the ID Token, I attempt to get the cached path from sessionStorage. On a failure I default to redirecting the user back to `/applications' to maintain the behavior that existed before this change. On a success I remove the item from sessionStorage to clean up resources then redirect the user to the cached path.

I use sessionStorage instead of a cookie as I do not need to send this information as part of an HTTP request. This path data does not leave the users browser and while I do add a step to clean up this resource, it is automatically cleaned up by the browser once the session is ended anyways.